server/api/batch_inventory.py

```diff
@@ -378,14 +384,82 @@ def validate_choice_name_and_get_choice_id(choice_name: str) -> Optional[str]:
                     )
 
                     batch_key: BatchKey = ("", batch)
-                    choice_id = validate_choice_name_and_get_choice_id(choice_name)
+                    choice_id = validate_choice_name_and_get_choice_id(
+                        contest, choice_name
+                    )
 
                     ballot_count_by_batch[batch_key] += 1
                     if choice_id:
                         batch_tallies[batch_key][choice_id] += 1
 
         # Set explicit zeros for choices with zero votes in a batch to avoid KeyErrors when
         # generating files
+        for batch_key in ballot_count_by_batch.keys():
+            if batch_key not in batch_tallies:
+                batch_tallies[batch_key] = defaultdict(int)
+        for tallies in batch_tallies.values():
+            for contest in contests:
+                for choice in contest.choices:
+                    if choice.id not in tallies:
+                        tallies[choice.id] = 0
+
+        election_results: ElectionResults = dict(
+            ballot_count_by_batch=dict_to_items_list(ballot_count_by_batch),
+            ballot_count_by_group=None,
+            batch_to_counting_group=None,
+            batch_tallies=dict_to_items_list(batch_tallies),
+        )
+        batch_inventory_data.election_results = election_results
+
+    def process_hart():
+        ballot_count_by_batch: Dict[BatchKey, int] = defaultdict(int)
+        batch_tallies: Dict[BatchKey, Dict[str, int]] = defaultdict(
+            lambda: defaultdict(int)
+        )
+
+        # cvr_file is a ZIP file with multiple XMLs
+        zip_entry_names = unzip_files(cvr_file, working_directory)
+        cvr_file_names = [
+            entry_name
+            for entry_name in zip_entry_names
+            if entry_name.lower().endswith(".xml") and not entry_name.startswith(".")
+            # ZIP files created on Macs include a hidden __MACOSX folder
+            and not entry_name.startswith("__")
```

Comment:
I don't see this check in our regular Hart CVR parsing. I wonder if it's unnecessary or if we should add it there as well

Reply:
will add it there